server/src/main/java/org/elasticsearch/index/query/SpanOrQueryBuilder.java

```diff
@@ -115,14 +129,16 @@ public static SpanOrQueryBuilder fromXContent(XContentParser parser) throws IOEx
                         if (query instanceof SpanQueryBuilder == false) {
                             throw new ParsingException(parser.getTokenLocation(), "spanOr [clauses] must be of type span query");
                         }
-                        clauses.add((SpanQueryBuilder) query);
+                        final SpanQueryBuilder clause = (SpanQueryBuilder) query;
```

Comment:
I think these 3 lines are not necessary, as each inner SpanQueryBuilder within their `fromXContent` should check for `boost` and fail if `boost` provided.

Reply:
Compound `span_or` queries can contain `span_term` inner queries and setting `boost` in `span_term` queries is allowed, that's why I've added this check here, do you agree with it? See this [test](https://github.com/elastic/elasticsearch/pull/34112/files#diff-9103f862a98e22ff9bc652661ba0a192R110).